extensions/notebook/src/book/bookTrustManager.ts

```diff
@@ -0,0 +1,161 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as path from 'path';
+import * as vscode from 'vscode';
+import * as constants from './../common/constants';
+import { BookTreeItem } from './bookTreeItem';
+import { BookModel } from './bookModel';
+
+export interface IBookTrustManager {
+	isNotebookTrustedByDefault(notebookUri: string): boolean;
+	setBookAsTrusted(bookRootPath: string): boolean;
+	setBookAsUntrusted(bookRootPath: string): boolean;
+}
+
+export interface IBookTrustManagerWorkspaceDetails {
+	workspaceFolders: vscode.WorkspaceFolder[];
+	getConfiguration(section?: string | undefined): vscode.WorkspaceConfiguration;
+}
+
+enum TrustBookOperation {
+	Add,
+	Remove
+}
+
+export class BookTrustManager implements IBookTrustManager {
+
+	private static notebookConfiguration: string = constants.notebookConfigKey;
+	private static notebookTrustedBooksConfiguration: string = constants.trustedBooksConfigKey;
+	private trustedLocalBooks: string[] = [];
+
+	constructor(private books: BookModel[], private workspaceDetails?: IBookTrustManagerWorkspaceDetails) {
+		if (!workspaceDetails) {
+			this.workspaceDetails = {
+				get getConfiguration() {
+					return vscode.workspace.getConfiguration;
+				},
+				get workspaceFolders() {
+					return [].concat(vscode.workspace.workspaceFolders || []);
+				}
+			};
+		}
+	}
+
+	isNotebookTrustedByDefault(notebookUri: string): boolean {
+		let allBooks: BookTreeItem[] = this.getAllBooks();
+		let trustableBookPaths = this.getTrustableBookPaths();
+		let trustableBooks: BookTreeItem[] = allBooks
+			.filter(bookItem => trustableBookPaths.some(trustableBookPath => trustableBookPath === path.join(bookItem.book.root, path.sep)));
```

Comment:
I keep seeing logic around adding and removing path separators which I'm a bit worried about - @MaddyDev already had to do some custom work around that to make the paths correctly open in editors and I feel like there's assumptions being made here that might not always be true. For example you use path.sep here, but is that always going to match what book.root has for its separators? The thing Maddy did involved switching to using posix paths so would always be /, even on Windows machines.

Have you tested this code on both mac/linux and windows?

Reply:
So yeah - you're getting some test failures on the linux build which seems to imply my concerns are valid here. Let's make sure to fully test this stuff on both platforms to verify it's working correctly.

Reply:
Oh I see, I dismissed it because I thought it was this build:
![image](https://user-images.githubusercontent.com/52225468/76120031-79725500-5fa5-11ea-9b0c-881d3f163218.png)

For some reason I thought the PR validation check was getting sign offs for the PR. Will look into this.

Reply:
#resolved